packages/cli/src/services/orchestration/main/handleCommandMessageMain.ts

```diff
@@ -131,6 +138,41 @@ export async function handleCommandMessageMain(messageString: string) {
 				await activeWorkflowRunner.removeActivationError(workflowId);
 				await activeWorkflowRunner.removeWorkflowTriggersAndPollers(workflowId);
 
+				push.broadcast('workflowDeactivated', { workflowId });
+
+				// instruct followers to show workflow deactivation in UI
+				await orchestrationService.publish('display-workflow-deactivation', { workflowId });
+
+				break;
+			}
+
+			case 'display-workflow-activation': {
+				if (!debounceMessageReceiver(message, 100)) {
+					message.payload = { result: 'debounced' };
+					return message;
+				}
+
+				const { workflowId } = message.payload ?? {};
+
+				if (typeof workflowId !== 'string') break;
+
+				push.broadcast('workflowActivated', { workflowId });
```

Comment:
Just to be sure, the UI is ready to only display this if the currently open workflow ID matches to the one sent in the payload right?

Reply:
The FE at `pushConnection.ts` receives this workflow ID and calls store methods:

```ts
if (receivedData.type === 'workflowActivated') {
	this.workflowsStore.setWorkflowActive(receivedData.data.workflowId);
	return true;
}

if (receivedData.type === 'workflowDeactivated') {
	this.workflowsStore.setWorkflowInactive(receivedData.data.workflowId);
	return true;
}
```

And the store methods compare the pushed workflow ID against `this.workflow.id`, which as far as I understand is the workflow on canvas.

```ts
setWorkflowActive(workflowId: string): void {
	const uiStore = useUIStore();
	uiStore.stateIsDirty = false;
	const index = this.activeWorkflows.indexOf(workflowId);
	if (index === -1) {
		this.activeWorkflows.push(workflowId);
	}
	if (this.workflowsById[workflowId]) {
		this.workflowsById[workflowId].active = true;
	}
	if (workflowId === this.workflow.id) {
		this.setActive(true);
	}
},

setWorkflowInactive(workflowId: string): void {
	const index = this.activeWorkflows.indexOf(workflowId);
	if (index !== -1) {
		this.activeWorkflows.splice(index, 1);
	}
	if (this.workflowsById[workflowId]) {
		this.workflowsById[workflowId].active = false;
	}
	if (workflowId === this.workflow.id) {
		this.setActive(false);
	}
},
```